src/DataValues/DecimalValue.php

```diff
@@ -360,13 +350,20 @@ public function getArrayValue() {
 	}
 
 	/**
-	 * Constructs a new instance of the DataValue from the provided data.
-	 * This can round-trip with @see getArrayValue
+	 * Constructs a new instance from the provided data. Required for @see DataValueDeserializer.
+	 * This is expected to round-trip with @see getArrayValue.
+	 *
+	 * @deprecated since 0.8.3. Static DataValue::newFromArray constructors like this are
+	 *  underspecified (not in the DataValue interface), and misleadingly named (should be named
+	 *  newFromArrayValue). Instead, use DataValue builder callbacks in @see DataValueDeserializer.
```

Comment:
I also don't say this is something more than a minor point. I just wanted to point out that this alternative suggested here is provided from the perspective of one particular use of this component.
As the same/very similar notice if now being suggested to be add to N other related components, I'd argue it is good to have an accurate documentation everywhere.
If others say the current notice is fine, I am not going to oppose. As I don't oppose here.